_core/smart/tutorial.md

```diff
@@ -0,0 +1,335 @@
+</p>
+</div>
+
+## Before you begin (pre-requisites)
+
+The Clarity language goes live in the next Stacks blockchain fork. Until the fork, you can run Clarity in a test environment. You run this test environment in a Docker container.  Before you begin this tutorial, make sure you have <a href="https://docs.docker.com" target="_blank">Docker installed on your workstation</a>.
+
+If for some reason you don't want to run the test environment with Docker, you can build and maintain a local environment. Instructions for downloading and building the environment are available in the `blockstack/blockstack-core` repository's <a href='https://github.com/blockstack/blockstack-core' target='_blank'>README</a> file.
+
+
+## Task 1: Set up the test environment
+
+Blockstack publishes an Clarity's developer preview image on Docker hub. This virtual environment (VM) that contains sample programs, the Blockstack Core, and tools for workign with them. In this task, you use Docker to pull and and run the image on your local workstation. 
+
+1. Pull the Blockstack core `clarity-developer-preview` image from Docker Hub.
+
+    ```bash
+    $ docker pull blockstack/blockstack-core:clarity-developer-preview
+    ``` 
+
+2. Start the Blockstack Core test environment with a Bash shell.
+
+    ```bash
+    $ docker run -it -v $HOME/blockstack-dev-data:/data/ blockstack/blockstack-core:clarity-developer-preview bash
+    ```
+
+    The launches a container with the Clarity VM and opens a bash shell into the VM. The `-v` flag creates a local `$HOME/blockstack-dev-data` directory in your workstation and mounts it at the `/data` directory inside the container. The shell opens into the `src/blockstack-core` directory. This directory contains the source for a core and includes Clarity contract samples you can run.
+
+3. List the contents of the `sample-programs` directory.
+
+   ```bash
+   root@f88368ba07b2:/src/blockstack-core# ls sample-programs/
+   names.clar  tokens.clar
+   ```
+
+   The sample programs directory contains two simple Clarity programs. Clarity code files have a `.clar` suffix.
+
+4. Go ahead and display the contents of the `tokens.clar` program with the `cat` command.
+
+    ```bash
+    root@c28600552694:/src/blockstack-core# cat sample-programs/tokens.clar 
+    ```
+
+    The next section gives you an introduction to the Clarity language by way of examining this program's code.
+
+## Task 2: Review a simple Clarity program
+
+If you haven't already done so, use the `cat` or `more` command to dipslay the `tokens.clar` file's code. Clarity is designed for static analysis; it is not a compiled language and is not Turing complete. It language is a LISP-like language.  LISP is an acronym for list processing. the first line of the program contains a user-defined `get-balance` function.  
```

Comment:
Typo: 'dipslay'

Reply:
I think it'd be helpful to say here that we're currently looking at the `tokens.clar` sample.

Reply:
Yep, I was aware of the duplication but some people just jump to the tutorial.

Reply:
Okay, works for me